src/ansi-c/expr2c.cpp

```diff
@@ -3494,9 +3494,18 @@ expr2ct::convert_extractbits(const extractbits_exprt &src, unsigned precedence)
 {
   std::string dest = convert_with_precedence(src.src(), precedence);
   dest+='[';
-  dest += convert_with_precedence(src.upper(), precedence);
+  auto expr_width_opt = pointer_offset_bits(src.type(), ns);
+  if(expr_width_opt.has_value())
+  {
+    auto upper = plus_exprt{
+      src.index(),
+      from_integer(expr_width_opt.value() - 1, src.index().type())};
+    dest += convert_with_precedence(upper, precedence);
+  }
+  else
+    dest += "?";
```

Comment:
Will this result in syntactically invalid C being generated?  If so, is that a problem and would it be better to throw some kind of error or have an invariant?

Reply:
The expr2X functions are designed to generate some string when given an unknown expression -- this should be added to the documentation.